optuna/pruners/threshold.py

```diff
@@ -0,0 +1,59 @@
+from optuna.pruners import BasePruner
+
+from optuna.type_checking import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from optuna.structs import FrozenTrial
+    from optuna.study import Study
+
+
+class ThresholdPruner(BasePruner):
+    """Pruner to detect abnormal metrics of the trials.
+
+    Prune if a metric exceeds upper bound threshold or
+    falls behind lower bound threshold which users specify.
```

Comment:
A nit, but I think it's slightly verbose.
```suggestion
    falls behind lower bound threshold.
```